src/components/sourcepublisher/SourcePublisher.js

```diff
@@ -52,43 +29,28 @@ function reset() {
   PubSub.publish(RESET, {});
 }
 
-function publishLayer(data, type, name, url) {
-  const schema = typeToSchema[type];
-  if (!schema) {
-    throw Error(`No schema for ${type}`);
-  }
-  const validate = new Ajv().compile(schema);
-  const valid = validate(data);
-  if (!valid) {
-    const failureReason = JSON.stringify(validate.errors, null, 2);
-    warn(`Error while validating ${name}. Details in console.`);
-    console.warn(`"${name}" (${type}) from ${url}: validation failed`, failureReason);
-  }
-  PubSub.publish(typeToEvent[type], { data, url });
-  info(name);
-}
-
 function loadLayer(layer) {
   const {
-    name, type, url, requestInit = {},
+    name, type, url,
   } = layer;
-  fetch(url, requestInit)
-    .then((response) => {
-      if (response.ok) {
-        response.json().then((data) => {
-          publishLayer(data, type, name, url);
-        }, (failureReason) => {
-          warn(`Error while parsing ${name}.`);
-          console.warn(`"${name}" (${type}) from ${url}: parse failed`, failureReason);
-        });
-      } else {
-        warn(`Error HTTP status from ${name}.`);
-        console.warn(`"${name}" (${type}) from ${url}: HTTP failed`, response.headers);
-      }
-    }, (failureReason) => {
-      warn(`Error while fetching ${name}.`);
-      console.warn(`"${name}" (${type}) from ${url}: fetch failed`, failureReason);
-    });
+
+  // Iterate over loaders in reverse priority order,
+  // and choose the best match based on the URL file extension.
```

Comment:
Or you could iterate in the opposite order, and exit on the first match?

Reply:
Good point, that does seem nicer, I have updated.